app/backend/lib/reporting/gcpe.ts

```diff
@@ -0,0 +1,592 @@
+                sowTab2SBySowId {
+                  totalCount
+                  nodes {
+                    jsonData
+                  }
+                }
+                sowTab7SBySowId {
+                  nodes {
+                    jsonData
+                  }
+                  totalCount
+                }
+                sowTab8SBySowId {
+                  totalCount
+                  nodes {
+                    jsonData
+                  }
+                }
+                jsonData
+              }
+              totalCount
+            }
+            ccbcNumber
+            externalStatus
+            internalDescription
+            package
+            projectName
+            status
+            intakeNumber
+            organizationName
+          }
+        }
+      }
+    }
+`;
+
+const createReportingGcpeMutation = `
+  mutation createReportingGcpe($input: CreateReportingGcpeInput!) {
+    createReportingGcpe(input: $input) {
+      clientMutationId
+      reportingGcpe {
+        rowId
+      }
+    }
+  }
+`;
+
+const getReportingGcpeQuery = `
+  query getReportingGcpe($rowId: Int!) {
+    reportingGcpeByRowId(rowId: $rowId) {
+      reportData
+      rowId
+    }
+  }
+`;
+
+export const regenerateGcpeReport = async (rowId, req) => {
+  const queryResult = await performQuery(
+    getReportingGcpeQuery,
+    { rowId: parseInt(rowId, 10) },
+    req
+  );
+  const excelData = queryResult.data.reportingGcpeByRowId.reportData;
+  // due to the way GraphQL mutation handles de/serialization, we need to reformat the data
+  excelData.forEach((row) => {
+    row.forEach((cell) => {
+      if (cell?.format === '$#,##0.00' || cell?.format === '0%') {
+        // eslint-disable-next-line no-param-reassign
+        cell.type = Number;
+      }
+    });
+  });
+  const blob = await writeXlsxFile(excelData as any, {
+    fontFamily: 'BC Sans',
+    fontSize: 12,
+    dateFormat: 'yyyy-mm-dd',
+    stickyColumnsCount: 7,
+    sheet: 'GCPE Report',
+    columns: columnOptions,
+  });
+  return blob;
+};
+
+const generateExcelData = async (
+  cbcData,
+  ccbcData,
+  compare = false,
+  prevExcelData?
+) => {
+  const infoRow = generateHeaderInfoRow();
+
+  const excelData = [infoRow, HEADER_ROW];
+
+  cbcData?.data?.allCbcData?.edges?.forEach(async (edges) => {
+    const { node } = edges;
+    const row: Row = [
+      // program
+      { value: 'CBC' },
+      // announced by province
+      { value: convertBoolean(node?.jsonData?.announcedByProvince) },
+      // change request pending
+      { value: convertBoolean(node?.jsonData?.changeRequestPending) },
+      // project complete
+      { value: node?.jsonData?.projectComplete },
+      // phase
+      { value: node?.jsonData?.phase },
+      // project #
+      { value: node?.projectNumber },
+      // ubf/federal project #
+      { value: node?.jsonData?.federalProjectNumber },
+      // applicant
+      { value: node?.jsonData?.applicantContractualName },
```

Comment:
Just to be sure, we're not using the current operating name here vs the contractual name?